WalletWasabi/CoinJoin/Coordinator/Coordinator.cs

```diff
@@ -221,23 +221,30 @@ public async Task MakeSureTwoRunningRoundsAsync(Money? feePerInputs = null, Mone
 
 				int confirmationTarget = await AdjustConfirmationTargetAsync(lockCoinJoins: true).ConfigureAwait(false);
 
+				var defaultTimeout = TimeSpan.FromSeconds(RoundConfig.InputRegistrationTimeout);
 				if (runningRoundCount == 0)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, defaultTimeout);
 					round.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					round.StatusChanged += Round_StatusChangedAsync;
 					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round);
 
-					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, 2 * defaultTimeout);
 					round2.StatusChanged += Round_StatusChangedAsync;
 					round2.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					await round2.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round2);
 				}
 				else if (runningRoundCount == 1)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					// If only one round is running then get its timeout
+					// if that's larger than the default timeout
+					// then we must make the new round's timeout smaller: existing round timeout / 2 for best spacing
+					// else larger: existing round timeout + default timeout
+					var existingRoundTimeout = Rounds.FirstOrDefault(x => x.Status == CoordinatorRoundStatus.Running)?.InputRegistrationTimeout ?? TimeSpan.Zero;
+					var timeout = existingRoundTimeout > defaultTimeout ? existingRoundTimeout / 2 : existingRoundTimeout + defaultTimeout;
```

Comment:
I'm not sure this is correct because one branch simply ignores the configured parameter. Imagine the current configuration ensures a round every hour then, I change it because I want to have a round every 10 minutes. We have an existing round with 1hr timeout and the new round is created with 30mins timeout. In case the existing round finished in 5 minutes the new created round will finish 25 minutes after that. 

Next time a round is created the timeout will not be 10 minutes either, it will be half of the existing one so, 15 minutes. Next one will be 7 minutes and then finally 17 minutes.

Reply:
This code ignores the change of the configured parameter, not the configured parameter itself. Fixed.